app/controllers/tweetscontroller.rb

```diff
@@ -0,0 +1,85 @@
+class TweetsController < ApplicationController
+
+    
+
+    # get '/show' do
+    #     @tweets = Tweet.all
+    #     erb :'/show'
+    # end
+
+    get '/tweets/new' do
+        if logged_in?
+             erb :'tweets/new'
+        else # logged out and doesn't let them view the create tweet
+
+        redirect to "/login"
+        end
+    end
+
+     post '/tweets' do
+        if params[:content] == "" #blank tweet
+            redirect to "/tweets/new"
+        else
+             @tweet = current_user.tweets.create(content: params[:content])
+             redirect to "/tweets/#{@tweet.id}"
+        end
+    end
+
+
+    get '/users/:slug' do
+        @tweets = Tweet.all
+        erb :'/tweets'
+    end
+
+    get '/tweets/:id' do
+        if logged_in?
+        @tweet = Tweet.find_by_id(params[:id])
+        erb :"/tweets/show"
+        else
+        redirect to '/login'
+        end
+    end
+
+   
+
+    get '/tweets/:id/edit' do
+        if logged_in?
+        @tweet = Tweet.find_by(params[:id])
+           erb :"/tweets/edit"
+           else
+        redirect to '/login'
+        end
+    end
+
+    patch '/tweets/:id' do
+        if logged_in?
+        @tweet = Tweet.find_by_id(params[:id])
+        @tweet.content = params[:content]
+            if params[:content] == ""
+                redirect to "/tweets/#{@tweet.id}/edit"
+            else
+                @tweet.save
+
+              redirect to "/tweets/#{@tweet.id}/edit"
+            end
+         else
+        redirect to "/login"
+        end
+    end
+ 
+
+    delete '/tweets/:id/delete' do
+        if logged_in?
+        @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.